encoding.go

```diff
@@ -0,0 +1,89 @@
+package verkle
+
+import (
+	"errors"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/rlp"
+)
+
+const ErrInvalidNodeEncoding = "invalid node encoding"
+
+func ParseNode(serialized []byte, tc *TreeConfig) (VerkleNode, error) {
+	elems, _, err := rlp.SplitList(serialized)
+	if err != nil {
+		return nil, err
+	}
+	c, err := rlp.CountValues(elems)
+	if err != nil {
+		return nil, err
+	}
+
+	if c == 1 {
+		// hashedNode
```

Comment:
I don't think it makes sense to save a `hashedNode` as the data is already present in the db and we can't really do anything with them. I understand this is connected to `insertOrdered` hashing previous nodes and thus causing most children in a node to be `hashedNode`s. This comforts me in my opinion, expressed in #30, that we should change `insertOrdered` not to hash leaves, and not save them to db when they are flushed. Parent nodes + its leaves should be committed to db together.